lib/perl/Genome/InstrumentData/AlignmentResult/Merged.pm

```diff
@@ -219,24 +226,70 @@ sub create {
         for my $alignment (@alignments) {
             $alignment->add_user(user => $self, label => 'uses');
         }
-
-        return 1;
-    };
-    if (my $error = $@) {
-        $tx->rollback();
-        die $error;
-    } elsif ($rv ne 1) {
+    }
+    catch {
         $tx->rollback();
-        $self->error_message('Unexpected return value: ' . $rv);
-        die $self->error_message;
-    } else {
-        $tx->commit();
+        die $self->error_message('Merge failed due to error: ' . $_);
     }
+    finally {
+        for my $allocation (@temp_allocations) {
+            $allocation->delete;
+        }
+        unless ($_) {
+            $tx->commit();
```

Comment:
It looks like based upon testing Try::Tiny in a script, you are correct. The commit will not run if something 'dies' in the delete.

It's possible to add multiple 'finally' blocks, and the LAST 'finally' to execute as part of this block needs a semicolon after its end brace.

Example for our own information:

```
my $var;
try {
    print "in the try\n";
    die "died in try";
    print "still printing in the TRY after death\n";
} catch {
    print "in the catch, got ($_)\n";
    die "died in catch ($_)";
    print "still printing in the CATCH after death\n";
} finally {
    $var = 1;
    print "in the finally, got ($_)\n";
    die "died in finally ($_)";
    print "still printing in the FINALLY after death\n";
} finally {
    $var = 2;
    print "in the SECOND finally, got ($_)\n";
}
finally {
    $var = 3;
    print "in the THIRD finally, got ($_)\n";
}

finally {
    $var = 4;
    print "in the FOURTH finally, got ($_)\n";
};

print "done with code, var is ($var)\n";`
```

```
in the try
in the catch, got (died in try at test.pl line 12.
)
died in catch (died in try at test.pl line 12.
) at test.pl line 16.
Use of uninitialized value $_ in concatenation (.) or string at test.pl line 34.
in the FOURTH finally, got ()
Use of uninitialized value $_ in concatenation (.) or string at test.pl line 29.
in the THIRD finally, got ()
Use of uninitialized value $_ in concatenation (.) or string at test.pl line 25.
in the SECOND finally, got ()
Use of uninitialized value $_ in concatenation (.) or string at test.pl line 20.
in the finally, got ()
Use of uninitialized value $_ in concatenation (.) or string at test.pl line 21.
        (in cleanup) died in finally () at test.pl line 21.
```